lib/src/repo.rs

```diff
@@ -1041,12 +1041,12 @@ impl MutableRepo {
     }
 
     fn update_all_references(&mut self, settings: &UserSettings) -> BackendResult<()> {
-        for (old_parent_id, rewrite) in self.parent_mapping.clone() {
-            // Call `new_parents()` here since `parent_mapping` only contains direct
+        for (old_parent_id, _rewrite) in self.parent_mapping.clone() {
+            // Call `rewritten_ids_with()` here since `parent_mapping` only contains direct
             // mappings, not transitive ones.
             // TODO: keep parent_mapping updated with transitive mappings so we don't need
-            // to call `new_parents()` here.
-            let new_parent_ids = self.new_parents(rewrite.new_parent_ids());
+            // to call `rewritten_ids_with()` here.
+            let new_parent_ids = self.rewritten_ids_with(slice::from_ref(&old_parent_id), |_| true);
```

Comment:
Seems useful to have a test for this, especially since I've apparently broken it once already. IIUC, `test_rebase_descendants_update_branches_after_divergent_rewrite` is close but in order to trigger this bug, we would need to add further rewrites (not necessarily divergent) after the first divergent rewrite.

Reply:
Added tests.

The behavior isn't exactly the same because intermediate rewrite doesn't appear in negative terms of the conflict, but I think that's okay.

Reply:
Looks good. Thanks!